runtime/compiler/z/codegen/J9CodeGenerator.cpp

```diff
@@ -4235,17 +4237,33 @@ J9::Z::CodeGenerator::inlineDirectCall(
 
    if (cg->getSupportsInlineStringIndexOf())
       {
-         switch (methodSymbol->getRecognizedMethod())
-            {
-            case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfLatin1:
-               resultReg = intrinsicIndexOf(node, cg, true);
+      switch (methodSymbol->getRecognizedMethod())
+         {
+         case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfLatin1:
+            resultReg = intrinsicIndexOf(node, cg, true);
+            return true;
+         case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfUTF16:
+            resultReg = intrinsicIndexOf(node, cg, false);
+            return true;
+         case TR::java_lang_StringLatin1_indexOf:
+         case TR::com_ibm_jit_JITHelpers_intrinsicIndexOfStringLatin1:
+            if (!comp->getOption(TR_DisableVectorStringIndexOfString))
```

Comment:
Don't we already have a `TR_DisableFastStringIndexOf` option for this? Do we need the new one?

Reply:
OK. I'll remove the new option and re-use `TR_DisableFastStringIndexOf`.